### PR TITLE
Prevent Herobox click when share this is displayed

### PR DESCRIPTION
**Pull request by 71**
**Issue:** #5628  
**Fix:** [Plunk](http://plnkr.co/edit/js6mbyl5lzwDtWoVXw1E?p=preview).

**Update:** Changed variable. However, I don't know (forgot some things about ng..) if, within the UserCtrl scope, we can access `spell, castEnd() and clickMember()`. If we can't, then my code won't work. Using the double ternary was the secure solution for me.

### CODE REVIEW

File: website/views/shared/avatar/appearance.jade [hunk 1]
```diff
@@ -1,6 +1,7 @@
 mixin avatar(opts)
   .character-sprites
     .addthis_native_toolbox(ng-if='::profile._id==user._id',
+      ng-mouseover='siHovered=true' ng-mouseout='siHovered=false' ng-init='siHovered=false'
```
Comment (crookedneighbor) [inline website/views/shared/avatar/appearance.jade]: This shouldn't require `ng-init` since `siHovered` will start out `undefined` which is a falsy value.

Comment (crookedneighbor) [inline website/views/shared/avatar/appearance.jade]: I would also be more explicit with the variable name. `shareIsHovered` would work for me.

File: website/views/shared/avatar/index.jade [hunk 2]
```diff
@@ -3,7 +3,7 @@ include appearance
 mixin herobox(opts)
   - if (!opts) {opts = {minimal:false,main:false}}
 
-  - var ngClick = 'spell ? castEnd(profile, "user", $event) : clickMember(profile._id)'
+  - var ngClick = 'spell ? castEnd(profile, "user", $event) : siHovered ? null : clickMember(profile._id)'
```
Comment (crookedneighbor) [inline website/views/shared/avatar/index.jade]: Once you get a double ternary in here, we should probably extract this to a function.

Comment (71) [inline website/views/shared/avatar/index.jade]: Then, I replace that by a function in the $scope, and I change the name?

**Comment by crookedneighbor** [comment]
This doesn't work as is (because of scoping issues), but even when I worked on the code a bit so that it _should_ work, it doesn't. Something about how the add this code works is preventing the mouse bindings from firing.

**Comment by crookedneighbor** [comment]
Going to close this for now, but when you're ready for me to review it again, please re-open it.

**Pull request closed without merging**